README.md

```diff
@@ -2,62 +2,75 @@
 
 This is a tool for taking screenshots of WMDE fundraising banners on wikipedia.org in different browsers and resolutions.
 
-## Configuration & Setup
+To improve performance, the system consists of parallelizable worker scripts for taking screenshots and processing metadata, connected by a message queue.
+
+## Configuration
 The screenshot background worker needs credentials for the Testingbot service. Put these in the file named `.env`.
 You can copy and adapt the file `env-template`.
 
-To install the dependencies run
+Make a copy of the `docker-compose.example-dev.file` named `docker-compose.dev.yml`. Edit the paths in the file to pount at the right files and directories:
 
-    npm install
+- The directory mounted to `/app/banner-shots` will contain the screenshots and metadata.
+- The file mounted to `/app/campaign_info.toml` must exist and contain a banner
+	configuration file (see below).
 
-or
+All calls to the `docker-compose` command must use the `-f` parameter with
+the two configuration files, creating an [override docker-compose
+configuration](https://docs.docker.com/compose/extends/#multiple-compose-files).
+Example:
 
-    docker run --rm -v $(pwd):/app -w /app node:16-alpine npm install
+	docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d
 
 ## Starting the Environment
 
 The screenshot tool needs background workers and a message queue (see architecture diagram below). To start these, run
 
-    docker-compose up -d
+	docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d
 
 This will start the background workers and [RabbitMQ](https://www.rabbitmq.com/) and expose it on Port 5672 on the 
 local machine.
 
 ## Creating Screenshots
 
-Run the screenshot tool with the following command
+### Downloading the campaign file for a branch
 
-    npx ts-node queue_screenshots.ts -c ../fundraising-banners/campaign_info.toml <CAMPAIGN_NAME>
-
-The `-c` parameter is for locating the campaign configuration from the
+The URL for the campaign configuration file from the
 [`wmde/fundraising-banners`
-repository](https://github.com/wmde/fundraising-banners).
-`<CAMPAIGN_NAME>` must be one of the configuration keys of that
-configuration file, e.g. `desktop` or `mobile`.
+repository](https://github.com/wmde/fundraising-banners) is
+`https://raw.githubusercontent.com/wmde/fundraising-banners/<BRANCH_NAME>/campaign_info.toml`.
+Replace the placeholder `<BRANCH_NAME>` with the name of the branch you
+want to use. To download the file you can use `wget` or `curl`. Examples:
+	
+	wget https://raw.githubusercontent.com/wmde/fundraising-banners/main/campaign_info.toml
 
-The background workers will create a directory inside the `banner-shots` directory. The campaign directory contains the 
-screenshot images and file `metadata.json` with all the metadata about the test case.
+	curl -o campaign_info.toml -L https://raw.githubusercontent.com/wmde/fundraising-banners/main/campaign_info.toml
```

Comment:
With our current setup, we'd almost never use main (since we want to test individual banner branches), so fetching the campaign info would look like 
  
    make BRANCH_NAME=C21_WMDE_Test_22 fetch-campaign-info

Still better than the long curl command, though.

The way I imagine our new workflow (each campaign in its own directory), we'd no longer branch and could always commit to main.